merge_report.py

```diff
@@ -0,0 +1,130 @@
+import sys
+import os
+import json
+import calendar
+import time
+
+import datetime
+import dateutil.relativedelta
+
+from github import Github
+from github import RateLimitExceededException
+import pandas as pd
+import argparse
+import pprint
+
+from github_connection.github import github_connection
+from rate_limiter.rate_limiter import rate_limiter
+from output.ouput import outputer
+from dataframe.csv import dataframe_to_csv
+from dataframe.markdown import dataframe_to_markdown
+
+from mergers.convertors import for_output
+from mergers.pull_requests import pull_requests
+
+pp = pprint.PrettyPrinter(indent=4)
+spacer = '      '
+
+
+
+# main class
+class merge_report(github_connection):
+    dataframe_handler = None
+    start_date = None
+    end_date = None
+    default_date_structure = {}
+
+    def __init__(self, organization_slug, team_slug, dataframe_handler, start_date, end_date):
+        super().__init__( organization_slug=organization_slug, team_slug=team_slug )
+        self.dataframe_handler = dataframe_handler
+        self.start_date = start_date
+        self.end_date = end_date
+
+        return
+
+    # main function body
+    def generate(self):
+        # conversion and save
+        out = outputer(for_output, self.dataframe_handler.save)
+        # get the team repos
+        print('---------------------------')
+        print('Getting team {} repos'.format(self.team_slug))
```

Comment:
you could use f" strings - much neater...
See: https://realpython.com/python-f-strings/
```suggestion
        print(f'Getting team {self.team_slug} repos')
```

Reply:
will give those a try going forward - not used them before